include/seastar/util/log-impl.hh

```diff
@@ -53,37 +53,32 @@ public:
     class inserter_iterator {
     public:
         using iterator_category = std::output_iterator_tag;
-        using difference_type = std::ptrdiff_t;
-        using value_type = char;
-        using pointer = char*;
-        using reference = char&;
+        using difference_type = void;
+        using value_type = void;
+        using pointer = void;
+        using reference = void;
```

Comment:
Why this change?

Reply:
1. I wanted to make it as close to `std` insert iterators as possible, and that's what they do.
2. `void`, `void`, `void`, `void` is the minimal set of typedefs. Why should we define the other types, if they are not used, and are not supposed to be used?

Reply:
I think I'll commit this patch as-is, but I'm leaving a few comments in case you want to consider them for follow-ups.
Here,  I don't understand what's the point of defining a type like "reference" if we define it to the "wrong" or "not supposed to be used" type. What forces us to define these aliases at all? Wouldn't it be better to leave them out completely, and if some code tries to use this "reference" they get an obvious error (because "reference" is not defined) instead of some subtle error because the type "void" doesn't fit whatever the user was trying to do with it?